src/lightcurve.jl

```diff
@@ -5,46 +5,70 @@ A structure representing a light curve, which is a time series of event counts.
 
 ## Fields
 
-- `timebins::Vector{T}`: Time bins for the light curve.
-- `counts::Vector{T}`: Number of events in each time bin.
-- `count_error::Vector{T}`: Error estimate for each bin.
-- `err_method::Symbol`: Method used for error estimation (`:poisson`).
+- timebins::Vector{T}: Time bins for the light curve.
+- counts::Vector{Int}: Number of events in each time bin.
+- count_error::Vector{T}: Error estimate for each bin.
+- err_method::Symbol: Method used for error estimation (:poisson).
 """
 struct LightCurve{T}
     timebins::Vector{T}
-    counts::Vector{T}
+    counts::Vector{Int}
     count_error::Vector{T}
     err_method::Symbol
+    
+    function LightCurve{T}(timebins, counts, count_error, err_method) where T
+        length(counts) == length(count_error) || throw(ArgumentError("Counts and error arrays must have the same length."))
+        return new(timebins, counts, count_error, err_method)
+    end
 end
 
 """
-    create_lightcurve(eventlist::EventList{T}, bin_size::T; err_method::Symbol=:poisson)
+    create_lightcurve(eventlist::EventList{T}, bin_size::T; err_method::Symbol=:poisson) where T
 
 Generate a light curve from an EventList by binning event times.
 
 ## Arguments
 
-- `eventlist::EventList{T}`: The input event list.
-- `bin_size::T`: The size of each time bin.
-- `err_method::Symbol`: The method for error estimation (`:poisson`).
+- eventlist::EventList{T}: The input event list.
```

Comment:
```suggestion
- `eventlist`: The input event list.
```
Docstrings should have things that refer to code in backticks, please add them back. You also don't need to annotate the type of parameters, since they are in the function signature.

Also, prefer using prose to describe the function and its arguments over lists. Have a look at some of the `Base` function documentation to get a feel for these things :+1:

Reply:
I've improved the docstring formatting.